# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoadContext.cpp
# Commit: e47abb2b8ac7
# Full Hash: e47abb2b8ac70efe89371c67766def82374c0f36
# Author: Alexandru Marc <amarc@mozilla.com>
# Date: 2025-09-27 12:29 +0300
# Regressor Bug: 1968890
# File Overlap Count: 14
# Overlapping Files: dom/base/test/jsmodules/chrome.toml, dom/base/test/jsmodules/test_dynamicImportErrorMessage.html, dom/script/ScriptLoadContext.cpp, dom/script/ModuleLoader.cpp, dom/base/test/jsmodules/import_404.mjs
# Description:
#   Revert "Bug 1968890 - Part 9: Add test for dynamic import. r=jonco" as potential cause of javascript engine crashes (bug 1991268)
# ==============================================================================

diff -r 7e2611c97326 -r e47abb2b8ac7 dom/script/ScriptLoadContext.cpp
--- a/dom/script/ScriptLoadContext.cpp	Sat Sep 27 11:41:54 2025 +0300
+++ b/dom/script/ScriptLoadContext.cpp	Sat Sep 27 12:29:56 2025 +0300
@@ -131,8 +131,7 @@
 }
 
 bool ScriptLoadContext::IsPreload() const {
-  if (mRequest->IsModuleRequest() &&
-      mRequest->AsModuleRequest()->IsStaticImport()) {
+  if (mRequest->IsModuleRequest() && !mRequest->IsTopLevel()) {
     JS::loader::ModuleLoadRequest* root =
         mRequest->AsModuleRequest()->GetRootModule();
     return root->GetScriptLoadContext()->IsPreload();